### PR TITLE
Suppress warnings for MD5 and SHA1 hashing function usages

### PR DESCRIPTION
**Pull request by leveyja**
## Description
Google Guava's `Hashing.md5()` and `Hashing.sha1()` are deprecated as insecure.
We use them to provide md5/sha1 functions, and therefore the usage is merited. 
This PR adds `@SuppressWarnings("deprecation")` to the necessary usages to avoid spurious deprecation warnings.

> Is this change a fix, improvement, new feature, refactoring, or other?

Refactoring

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Code change to suppress build warnings / no functional changes (apart from 1 test)

> How would you describe this change to a non-technical end user or system administrator?

Resolve build warnings

## Documentation

(:white_check_mark:) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(:white_check_mark:) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/operator/scalar/RowToRowCast.java [hunk 1]
```diff
@@ -165,6 +165,7 @@ private static Class<?> generateRowCast(Type fromType, Type toType, FunctionDepe
 
         // Embed the MD5 hash code of input and output types into the generated class name instead of the raw type names,
         // which could prevent the class name from hitting the length limitation and invalid characters.
+        @SuppressWarnings("deprecation")
         byte[] md5Suffix = Hashing.md5().hashBytes((fromType + "$" + toType).getBytes(UTF_8)).asBytes();
```
Comment (electrum) [inline core/trino-main/src/main/java/io/trino/operator/scalar/RowToRowCast.java:169]: This is internal and doesn't need to use MD5, so we could switch to `goodFastHash`

**Review by electrum — APPROVED** [review]

File: plugin/trino-raptor-legacy/src/test/java/io/trino/plugin/raptor/legacy/storage/TestRaptorStorageManager.java [hunk 2]
```diff
@@ -628,7 +629,8 @@ public static RaptorStorageManager createRaptorStorageManager(
     private static void assertFileEquals(File actual, File expected)
             throws IOException
     {
-        assertEquals(hash(actual, md5()), hash(expected, md5()));
+        HashFunction func = Hashing.goodFastHash(32);
```
Comment (electrum) [inline plugin/trino-raptor-legacy/src/test/java/io/trino/plugin/raptor/legacy/storage/TestRaptorStorageManager.java:632]: Let's switch this to
```java
assertThat(actual).hasSameBinaryContentAs(expected);

Comment (leveyja) [reply plugin/trino-raptor-legacy/src/test/java/io/trino/plugin/raptor/legacy/storage/TestRaptorStorageManager.java]: Nice to know this exists! Thanks

File: core/trino-main/src/main/java/io/trino/operator/scalar/RowToRowCast.java (hunk 1 above)
Comment (leveyja) [reply core/trino-main/src/main/java/io/trino/operator/scalar/RowToRowCast.java]: Created https://github.com/trinodb/trino/pull/12650

**Comment by leveyja** [comment]
Looks ready to merge - any suggestions who should give the final 👍 ?

**Pull request merged**